protocols/ping/src/handler.rs

```diff
@@ -288,19 +275,12 @@ impl ConnectionHandler for Handler {
 
                 self.failures += 1;
 
-                // Note: For backward-compatibility, with configured
-                // `max_failures == 1`, the first failure is always "free"
-                // and silent. This allows peers who still use a new substream
+                // Note: For backward-compatibility the first failure is always "free"
+                // and silent. This allows peers who use a new substream
```

Comment:
Is this backwards compatibility still relevant? This implementation should be compatible with any recent other implementation adhering to the [specification](https://github.com/libp2p/specs/blob/master/ping/ping.md), right?

Reply:
We could reword it but the functionality still needs to be there. JS for example uses a new stream per ping and we should not report that as an error.